TTCN3_Project/javasrc/CANPortFilter/CANPortFilter.java

```diff
@@ -0,0 +1,84 @@
+package CANPortFilter;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.etsi.ttcn.tri.TriAddress;
+import org.etsi.ttcn.tri.TriComponentId;
+import org.etsi.ttcn.tri.TriMessage;
+import org.etsi.ttcn.tri.TriPortId;
+import org.etsi.ttcn.tri.TriStatus;
+
+import com.testingtech.ttcn.tri.PortFilter;
+
+public class CANPortFilter extends PortFilter {
+	
```

Comment:
The following global variables are only used in `triEnqueueMsg()`, so they should be local variables.